Dockerfile: Update CLI, buildx and compose

vvoland opened the pull request:
### Dockerfile: update docker CLI to v27.2.1

Update the Docker CLI used in the dev-container

full diff: https://github.com/docker/cli/compare/v27.0.2...v27.2.1

### Dockerfile: update buildx to 0.17.0

Update the buildx cli plugin used in the dev-container

full diff: https://github.com/docker/buildx/compare/0.16.1...0.17.0

### Dockerfile: update compose to v2.29.2

Update the compose cli plugin used in the dev-container

full diff: https://github.com/docker/cli/compare/v2.29.0...v2.29.2

vvoland committed:
Dockerfile: update docker CLI to v27.2.1

Update the Docker CLI used in the dev-container

full diff: https://github.com/docker/cli/compare/v27.0.2...v27.2.1

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
Dockerfile: update buildx to 0.17.0

Update the buildx cli plugin used in the dev-container

full diff: https://github.com/docker/buildx/compare/0.16.1...0.17.0

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

thaJeztah commented on Dockerfile:
```diff
@@ -8,11 +8,11 @@ ARG XX_VERSION=1.5.0
 ARG DOCKERCLI_INTEGRATION_VERSION=v17.06.2-ce
-ARG BUILDX_VERSION=0.16.1
+ARG BUILDX_VERSION=0.17.0
 ARG COMPOSE_VERSION=v2.29.0
```
Compose can also be updated to v2.29.2; https://github.com/docker/compose/releases/tag/v2.29.2

thaJeztah reviewed (approved):
LGTM

left a suggestion for compose

vvoland committed:
Dockerfile: update compose to v2.29.2

Update the compose cli plugin used in the dev-container

full diff: https://github.com/docker/cli/compare/v2.29.0...v2.29.2

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

thaJeztah reviewed (approved):
LGTM

thaJeztah commented:
Failure is unrelated and tracked in https://github.com/moby/moby/issues/48400
